lib/plugins/threaded-signing/plugin.c

```diff
@@ -188,25 +226,17 @@ threaded_openssl_sign_hash(sv_threaded_plugin_t *self, const signature_info_t *s
   SignedVideoReturnCode status = SV_UNKNOWN_FAILURE;
   g_mutex_lock(&self->mutex);
 
-  // If the signature has not yet been pulled, or even generated, a new signature cannot be
-  // generated without replacing it. Log in |nbr_of_unsigned_hashes| and move to done.
-  if (self->plugin_state != THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN) {
-    self->nbr_of_unsigned_hashes++;
-    status = SV_OK;
-    goto done;
-  }
-
   // If no |self->signature_info| exists. Allocate necessary memory for it and copy the
   // |private_key| and |algo|.
   if (!self->signature_info) {
     self->signature_info = local_signature_info_create(signature_info);
     if (!self->signature_info) goto catch_error;
   }
 
-  // Allocate memory for the |hash_to_sign| if necessary.
-  if (!self->hash_to_sign) {
-    self->hash_to_sign = calloc(1, signature_info->hash_size);
-    if (!self->hash_to_sign) goto catch_error;
+  // Allocate memory for the |input_buffer| if necessary.
```

Comment:
To avoid all these malloc + free after consumption there are (at least) two other approaches to solve it.

1. Still allocate memory for a hash when there is no available slot, but instead of freeing it on line 164 you can move the pointer to the first non-NULL element in `input_buffer`
2. Use one big memory to stack all the hashes and use `memmove()` on the buffer after fetching a hash for signing